Content.Shared/Paint/PaintedComponent.cs

```diff
@@ -0,0 +1,35 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Paint;
+
+/// <summary>
+/// Component applied to target entity when painted.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class PaintedComponent : Component
+{
+    /// <summary>
+    ///  Color of the paint.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public Color Color = Color.FromHex("#2cdbd5");
+
+    /// <summary>
+    ///  Used to remove the color when component removed.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public Color BeforeColor;
+
+    /// <summary>
+    /// If paint is enabled.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool Enabled;
+
+    /// <summary>
+    /// Name of the shader.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public string ShaderName = "Greyscale";
+}
```

Comment:
Doesn't this need to store per-layer colors + shader?

It also needs to be able to support other systems modifying it and just not overriding the data.

Reply:
Not really sure how to store per layer colors and shaders. I assume through metadata?

Reply:
colors do not need to be stored, seems when i color the shader it does not affect the color being reverted and it goes back to its original coloration.
This only doesnt work with entites which have RandomShaderComponent (i just prevent painting of entites with this component for now). When removing the color we check that color is same as component.

If a layer has a shader already then we do not replace the shader or color (9/10 it will be a light of some sort). These means we never need to store the original shader data (shaders are hardly used and layers with shaders typically shouldnt be spray painted anyway). When removing the shader we now check that the shader matches the component (incase it was modified post painting).